snippets/components/video-render.js

```diff
@@ -0,0 +1,147 @@
+class LazyVideoComponent extends HTMLElement {
+  constructor() {
+    super();
+
+    const width = this.getAttribute("width");
+    const height = this.getAttribute("height");
+
+    const aspectRatio = this.getAspectRatio(width, height);
+
+    this.videoElement = document.createElement("template");
+    this.videoElement.innerHTML = `
+        <style>
+          video {
+            max-width: 100%;
+            aspect-ratio: ${aspectRatio};
+          }
+        </style>
+        
+        <video muted autoplay></video>  
+      `;
+
+    if (!this.shadowRoot) {
+      this.attachShadow({ mode: "open" });
+      this.shadowRoot.append(this.videoElement.content.cloneNode(true));
+    }
+
+    this._isLoaded = false;
+    this.observer = null;
+    this.update();
+  }
+
+  connectedCallback() {
+    this.setObserver();
+  }
+
+  disconnectedCallback() {
+    this.observer?.disconnect();
+  }
+
+  get isLoaded() {
+    return this._isLoaded;
+  }
+
+  set isLoaded(value) {
+    this._isLoaded = value;
+  }
+
+  setObserver() {
+    this.observer = new IntersectionObserver((entries) => {
+      const video = this.shadowRoot.querySelector("video");
+
+      entries.forEach((entry) => {
+        if (entry.isIntersecting) {
+          this.handleVideoInView(video);
+        } else {
+          this.handleVideoOutOfView(video);
+        }
+      });
+    });
+
+    this.observer.observe(this);
+  }
+
+  handleVideoInView(video) {
+    if (!this.isLoaded) {
+      this.loadVideo(video);
+      this.observer.unobserve(video);
+    } else {
+      video.play();
+    }
+  }
+
+  handleVideoOutOfView(video) {
+    if (this.isLoaded) {
+      video.pause();
+    }
+  }
+
+  loadVideo(video) {
+    Array.from(video.children).forEach((child) => {
+      if (child.tagName === "source".toUpperCase()) {
+        child.src = child.getAttribute("data-src");
+      }
+    });
+
+    video.load();
+    this.isLoaded = true;
+  }
+
+  update() {
```

Comment:
```suggestion
  createVideoSources() {
```

Small naming change that I think would make this clearer to understand in the long run.